public/app/features/alerting/unified/components/rule-editor/AlertTypeStep.tsx

```diff
@@ -0,0 +1,176 @@
+
+const alertTypeOptions: SelectableValue[] = [
+  {
+    label: 'Threshold',
+    value: RuleFormType.threshold,
+    description: 'Metric alert based on a defined threshold',
+  },
+  {
+    label: 'System or application',
+    value: RuleFormType.system,
+    description: 'Alert based on a system or application behavior. Based on Prometheus.',
+  },
+];
+
+export const AlertTypeStep: FC = () => {
+  const styles = useStyles(getStyles);
+
+  const { register, control, watch, errors, setValue } = useFormContext<RuleFormValues>();
+
+  const ruleFormType = watch('type');
+  const dataSourceName = watch('dataSourceName');
+
+  useEffect(() => {}, [ruleFormType]);
+
+  const rulesSourcesWithRuler = useRulesSourcesWithRuler();
+
+  const dataSourceFilter = useCallback(
+    (ds: DataSourceInstanceSettings): boolean => {
+      if (ruleFormType === RuleFormType.threshold) {
+        return !!ds.meta.alerting;
+      } else {
+        // filter out only rules sources that support ruler and thus can have alerts edited
+        return !!rulesSourcesWithRuler.find(({ id }) => id === ds.id);
+      }
+    },
+    [ruleFormType, rulesSourcesWithRuler]
+  );
+
+  return (
+    <RuleEditorSection stepNo={1} title="Alert type">
+      <Field
+        className={styles.formInput}
+        label="Alert name"
+        error={errors?.name?.message}
+        invalid={!!errors.name?.message}
+      >
+        <Input
+          autoFocus={true}
+          ref={register({ required: { value: true, message: 'Must enter an alert name' } })}
+          name="name"
+        />
+      </Field>
+      <div className={styles.flexRow}>
+        <Field
+          label="Alert type"
+          className={styles.formInput}
+          error={errors.type?.message}
+          invalid={!!errors.type?.message}
+        >
+          <InputControl
+            as={Select}
+            name="type"
+            options={alertTypeOptions}
+            control={control}
+            rules={{
+              required: { value: true, message: 'Please select alert type' },
+            }}
+            onChange={(values: SelectableValue[]) => {
+              const value = values[0]?.value;
+              // when switching to system alerts, null out data source selection if it's not a rules source with ruler
+              if (
+                value === RuleFormType.system &&
+                dataSourceName &&
+                !rulesSourcesWithRuler.find(({ name }) => name === dataSourceName)
+              ) {
+                setValue('dataSourceName', null);
+              }
+              return value;
+            }}
+          />
+        </Field>
+        <Field
+          className={styles.formInput}
+          label="Select data source"
+          error={errors.dataSourceName?.message}
+          invalid={!!errors.dataSourceName?.message}
+        >
+          <InputControl
+            as={DataSourcePicker as React.ComponentType<Omit<DataSourcePickerProps, 'current'>>}
```

Comment:
We'll have data source pickers on each query row so it doesn't make sense to have one here too.

Reply:
Lots of people bring this up :D but UX wanted to have it anyway. Let's dicuss on tomorrows UX meeting?

Reply:
Yes!